src/nvim/api/vim.c

```diff
@@ -654,12 +654,13 @@ Object nvim_get_var(String name, Error *err)
 {
   dictitem_T *di = tv_dict_find(&globvardict, name.data, (ptrdiff_t)name.size);
   if (di == NULL) {  // try to autoload script
-    VALIDATE_S((script_autoload(name.data, name.size, false) && !aborting()), "key", name.data, {
+    bool found = script_autoload(name.data, name.size, false) && !aborting();
+    VALIDATE(found, "Key not found: %s", name.data, {
```

Comment:
This should really be something like `"Invalid global var"`, but that looks like a deep change (not just this function) to be consistent across all the var-setting API functions.